src/gpu/nvidia/cudnn_batch_normalization_executor.hpp

```diff
@@ -40,155 +41,161 @@ struct bnorm_exec_base_t {
         });
     }
 
-    template <typename T>
-    void maybe_init_ss(
-            nvidia::sycl_cuda_stream_t *cuda_stream, T, T, bool) const {}
-
-    template <typename T>
-    void maybe_init_ss(nvidia::sycl_cuda_stream_t *cuda_stream,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> scale_acc,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> bias_acc,
-            bool init_ss) const {
-        if (init_ss) {
-            constexpr T scale_val = 1, bias_val = 0;
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(scale_acc, scale_val);
-            });
-
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(bias_acc, bias_val);
-            });
-        }
+    template <typename T = float>
+    void init_scaleshift(cuda_sycl_scoped_context_handler_t &sc,
+            const compat::interop_handle &ih,
+            nvidia::sycl_cuda_stream_t *cuda_stream, T, size_t) const {}
+
+    template <typename T = float, typename U>
+    void init_scaleshift(cuda_sycl_scoped_context_handler_t &sc,
+            const compat::interop_handle &ih,
+            nvidia::sycl_cuda_stream_t *cuda_stream,
+            sycl_memory_arg<::sycl::access::mode::write, U> arg_scale_bias,
+            const size_t n) const {
+        T scale_val = 1, bias_val = 0;
+        cuda_stream->interop_task([&](::sycl::handler &cgh) {
+            T *scale_ptr = static_cast<T *>(
+                    arg_scale_bias.get_native_pointer(ih, sc));
+            T *bias_ptr = scale_ptr + n;
+            cuMemsetD32Async(reinterpret_cast<CUdeviceptr>(scale_ptr),
+                    reinterpret_cast<int &>(scale_val), n,
+                    cuda_stream->get_underlying_stream());
+            cuMemsetD32Async(reinterpret_cast<CUdeviceptr>(bias_ptr),
+                    reinterpret_cast<int &>(bias_val), n,
+                    cuda_stream->get_underlying_stream());
+            cudaDeviceSynchronize();
+        });
     }
 
     // Handle the cases when mean and var are read-only accessors or nullptr
-    template <typename T>
-    void maybe_init_mean_var(
-            nvidia::sycl_cuda_stream_t *cuda_stream, T, T, bool) const {}
-
-    template <typename T>
-    void maybe_init_mean_var(nvidia::sycl_cuda_stream_t *cuda_stream,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> mean_acc,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> var_acc,
-            bool init_mean_var) const {
-        if (init_mean_var) {
-            constexpr T mean_var_val = 0;
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(mean_acc, mean_var_val);
-            });
-
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(var_acc, mean_var_val);
-            });
-        }
+    template <typename T, typename U>
+    void init_mean_var(cuda_sycl_scoped_context_handler_t &sc,
```

Comment:
I'm not sure that we have the cases that should be handled by this function. Am I missing something?

Reply:
Empty functions allow for other cases. See above comment